### ISSUE TITLE
Transition from comma to pipe separated values

### ISSUE THREAD

**Issue by riceissa** [vipulnaik/donations#49]
- [x] Review all schemas and come up with a list of fields where the transition may have to be made. List so far: https://github.com/vipulnaik/donations/blob/304571a5748ee12a82e739d3cded19d91166a6d3/sql/documents/documents-schema.sql#L9-L10 and https://github.com/vipulnaik/donations/blob/304571a5748ee12a82e739d3cded19d91166a6d3/sql/donations/donations-schema.sql#L26-L29 and https://github.com/vipulnaik/donations/blob/304571a5748ee12a82e739d3cded19d91166a6d3/sql/donors/donors-schema.sql#L7 and https://github.com/vipulnaik/donations/blob/304571a5748ee12a82e739d3cded19d91166a6d3/sql/money_moved.sql#L4-L5
- [x] Update the PHP rendering code to parse using pipe instead of comma. For instance, the code that populates the Affected donors and Affected donees columns at https://donations.vipulnaik.com/donor.php?donor=Open+Philanthropy+Project#donorDocumentList
- [x] Review all manually written SQL files and update them.
- [x]  Reinstate commas in donor and donee names that were elided because of this issue.
- [x]  Fix automated scripts as well.

**Comment by riceissa** [comment]
It looks like `author` in documents is also a CSV

**Comment by riceissa** [comment]
I'm working with this restricted list of SQL files which I *think* excludes all the script-generated SQL files: https://gist.github.com/riceissa/cffe01a35f92ce9d060ff0c89ad484a8

**Comment by riceissa** [comment]
So I'm doing the "Review all manually written SQL files and update them" step right now. My current process looks like this:

1. I run a query like `select affected_donors from documents where affected_donors REGEXP ',';` to find all the places a comma appears in one of the fields listed above.
2. I search in the SQL for where the string appears, and replace the commas with pipes.

Once I run out of commas to replace, I will go and insert in commas to entities (like "80,000 Hours") that require them.

**Comment by riceissa** [comment]
Queries to run:

- [ ] `select affected_donors from documents where affected_donors REGEXP ',';`
- [ ] `select affected_donees from documents where affected_donors REGEXP ',';`

**Comment by riceissa** [comment]
```
mysql> select distinct(dlw_data_processing_script) from donors;
+--------------------------------------------------------------+                        
| dlw_data_processing_script                                   |                        
+--------------------------------------------------------------+
| NULL                                                         |                        
| https://github.com/riceissa/michaeldello-donations           |                        
| https://github.com/riceissa/gates-foundation-iati-processing |                        
| https://github.com/riceissa/arnold-foundation-grants         |                        
| https://github.com/riceissa/public-welfare-foundation        |                        
| https://github.com/riceissa/ea-grants-processing             |                        
| https://github.com/riceissa/macarthur-foundation             |                        
| https://github.com/riceissa/hewlett-foundation               |                        
| https://github.com/riceissa/ford-foundation                  |                        
| https://github.com/riceissa/wellcome-trust                   |                        
| https://github.com/riceissa/bauman-foundation                |                        
| https://github.com/riceissa/unbound-philanthropy             |                        
| https://github.com/riceissa/templeton-foundation             |                        
| https://github.com/riceissa/mcdowell-foundation              |                        
| https://github.com/riceissa/nathan-cummings-foundation       |                        
| https://github.com/riceissa/pineapple-fund                   |                        
+--------------------------------------------------------------+                        
16 rows in set (0.00 sec)
```

**Comment by riceissa** [comment]
Ok, I looked through all of the processing scripts listed above, and it looks like none of them removed the comma in the first place (oops), which means less work now (yay).

**Comment by riceissa** [comment]
Other things to check:

- [x] https://github.com/vipulnaik/donations/tree/master/dedup-donees
- [ ] sloan
- [ ] mulago
- [ ] knight
- [ ] gates
- [ ] FLI
- [ ] fqxi
- [ ] EA survey
- [ ] CEA donor lottery
- [ ] open phil disclosures

**Comment by riceissa** [comment]
So actually for a lot of the scripts, I've mostly been using the location information verbatim, which means some of them might need to be split into multiple locations (and separated using pipes).

**Comment by riceissa** [comment]
Social media mass grab touches neither `key_people` nor `recommenders` for donees so nothing to do there.

**Comment by riceissa** [comment]
Nathan Cummings and [Gates IATI](https://github.com/riceissa/gates-foundation-iati-processing) need to be regenerated because the SQL file is now different.

Otherwise, this is basically done.

**Comment by riceissa** [comment]
By the way, the code changes from this issue (that are part of this repo) are in https://github.com/vipulnaik/donations/tree/pipe-separated-values

**Comment by vipulnaik** [comment]
I have merged the changes in, but have not regenerated the entire SQL (just wanted to merge them in so that I can add stuff to the SQL without creating merge conflicts).

@riceissa, can you take care of regenerating the data for Gates and Cummings and copying the regenerated data over to the relevant locations in the donations repo?

Thanks!

**Comment by riceissa** [comment]
Ok done (see 3368750a91706c115735ba4f1109325cb09e8903 and 095070871eaa68e39b6a925f0311529679421f05).

**Comment by vipulnaik** [comment]
@riceissa I think we missed the `influencer` column in the donations table.

**Comment by riceissa** [comment]
@vipulnaik You're right. Here are the ones I see https://gist.github.com/riceissa/bb129cff0cae8f0989928842bc5dfe63

**Comment by riceissa** [comment]
I fixed the ones listed above in 1bb2d48fb68ecd18ceec0380a3ede08ff50e421f

Now checking the PHP code.

I don't think the data generation code needs to be checked because all the influencer was never touched. (All the examples flagged in the above Gist are from manual entry files.)

**Comment by riceissa** [comment]
PHP code fix is done in 7e530b66847f45532f043d45d94a152dd2bd6025.

**Comment by riceissa** [comment]
@vipulnaik So I think this is done again, unless you find some other thing.
